### PR TITLE
Updated info for lifecycle events

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Updated info for lifecycle events

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 1]
```diff
@@ -196,6 +191,64 @@ RudderConfigBuilder configBuilder = new RudderConfigBuilder()
...
 ```
 
+## Tracking application lifecycle events for Android platform
+
+The Unity SDK tracks and sends the application lifecycle events to get insights into the app-related metrics like installs, opens, updates, etc. However, in order to track the application life cycle events on the Android platform, you need to add the `RudderPreferbs.prefab` file in the `Assets/Rudder/RudderPreferbs.prefab` path to each and every scene in your Unity app. Also, make sure that the `RudderPreferbs.prefab` is linked to the `RudderClient.cs` script.
+
+Unity SDK depends on the life cycle method `onApplicationFocus` of the MonoBehaviour to determine the **Application Opened** and **Application Backgrounded** events on the Android platform. Hence, when an application is brought to focus, an **Application Opened** event is sent and 
+when the application is moved out of focus, an **Application Backgrounded** event is sent. 
+
+So, these events might be triggered even before the Rudder SDK gets initialized so as to create actions and execute them once the SDK is initialized.
```
Comment (desusai7) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:201]:
```suggestion
So, these events might be triggered even before the Rudder SDK gets initialized so we create actions for them and execute them once the SDK is initialized.
```

**Review by desusai7 — APPROVED** [review]

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 2]
```diff
@@ -196,6 +191,64 @@ RudderConfigBuilder configBuilder = new RudderConfigBuilder()
     .withAutoCollectAdvertId(true);
 ```
 
+## Tracking application lifecycle events for Android platform
+
+The Unity SDK tracks and sends the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app-related metrics like installs, opens, updates, etc. However, in order to track the application life cycle events on the Android platform, you need to add the `RudderPreferbs.prefab` file from the path `Assets/Rudder/RudderPreferbs.prefab` to each and every scene in your Unity app. Also, make sure that the `RudderPreferbs.prefab` is linked to the `RudderClient.cs` script.
```
Comment (arnab-p) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:196]:
```suggestion
The Unity SDK tracks and sends the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app metrics like installs, opens, updates, etc. However, in order to track the application life cycle events on the Android platform, you need to add the `RudderPreferbs.prefab` file from the path `Assets/Rudder/RudderPreferbs.prefab` to every scene in your Unity app. Also, ensure that the `RudderPreferbs.prefab` is linked to the `RudderClient.cs` script.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 3]
```diff
@@ -196,6 +191,64 @@ RudderConfigBuilder configBuilder = new RudderConfigBuilder()
...
+
+    if (isSDKInitialized) {
+      RudderLogger.LogDebug("SDK Already initialized, executing the actions directly");
+      action();
+    } else {
+      RudderLogger.LogDebug("SDK not initialized yet, adding the actions to the list");
+      actionsList.Add(action);
+    }
+  }
+}
```
Comment (arnab-p) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:235]: We don't need to put the code in the documentation.

**Review by arnab-p — CHANGES_REQUESTED** [review]
We should mention the `trackLifecycleEvent` config clearly and say that we default it to `true`

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 4]
```diff
@@ -196,6 +191,39 @@ RudderConfigBuilder configBuilder = new RudderConfigBuilder()
...
+
+The Unity SDK automatically tracks the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app metrics like installs, opens, updates, etc. However, you can disable the automatic tracking by using the following code:
+
+```csharp
+RudderConfig config = new RudderConfigBuilder()
+  .WithTrackLifecycleEvents(false)
+  .WithDataPlaneUrl("DATA_PLANE_URL")
+  .WithLogLevel(RudderLogLevel.VERBOSE)
+  .Build();
+rudderClient = RudderClient.GetInstance("WRITE_KEY", config);
```
Comment (arnab-p) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:204]: Let's only mention about the `  .WithTrackLifecycleEvents(false)` part. With this code snippet, the reader has to look for the change they have to make to their existing codebase.

File: rudderstack-api/api-specification/application-lifecycle-events-spec.mdx [hunk 5]
```diff
@@ -9,11 +9,13 @@ description: >-
 
 RudderStack lets you track various application lifecycle events across the mobile SDKs and get insights into app-related metrics like installs, opens, etc. This guide provides the details and semantic definitions of these events and the associated properties.
 
-<div class="warningBlock">
-
-The Unity SDK currently does not support tracking the application lifecycle events on the Android platform.
+<div class="infoBlock">
+  
+To track the application lifecycle events on Android platform using the Unity SDK, you need to perform some additional steps. For more information on these steps, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk/#tracking-application-lifecycle-events-for-android-platform">Tracking application lifecycle events for Android platform</a> section.
```
Comment (ameypv-rudder) [inline rudderstack-api/api-specification/application-lifecycle-events-spec.mdx:14]:
```suggestion
To track the application lifecycle events on the Android platform using the Unity SDK, you need to perform some additional steps. For more information on these steps, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk/#tracking-application-lifecycle-events-for-android-platform">Tracking application lifecycle events for the Android platform</a> section.
```

File: rudderstack-api/api-specification/application-lifecycle-events-spec.mdx [hunk 6]
```diff
@@ -154,6 +153,13 @@ RudderStack tracks the following properties for this event:
 | `previous_version` | String | The application version before it was updated.  |
 | `version` | String | The current version of the application after it is updated. |
 
+
+<div class="infoBlock">
+
+For Unity SDK, you need to perform the required steps for Android and iOS platforms to trigger the Application Updated lifecycle event. For more information on these steps, refer to the
```
Comment (ameypv-rudder) [inline rudderstack-api/api-specification/application-lifecycle-events-spec.mdx:159]:
```suggestion
For the Unity SDK, you need to perform the required steps for the Android and iOS platforms to trigger the <strong>Application Updated</strong> lifecycle event. For more information on these steps, refer to the 
```

File: rudderstack-api/api-specification/application-lifecycle-events-spec.mdx [hunk 7]
```diff
@@ -9,11 +9,13 @@ description: >-
 
 RudderStack lets you track various application lifecycle events across the mobile SDKs and get insights into app-related metrics like installs, opens, etc. This guide provides the details and semantic definitions of these events and the associated properties.
 
-<div class="warningBlock">
-
-The Unity SDK currently does not support tracking the application lifecycle events on the Android platform.
+<div class="infoBlock">
+  
+To track the application lifecycle events on the Android platform using the Unity SDK, you need to perform some additional steps. For more information on these steps, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk/#tracking-application-lifecycle-events-for-android-platform">Tracking application lifecycle events for the Android platform</a> section.
```
Comment (ameypv-rudder) [inline rudderstack-api/api-specification/application-lifecycle-events-spec.mdx:14]:
```suggestion
To track the application lifecycle events on the Android platform using the Unity SDK, you need to perform some additional steps. For more information on these steps, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk/#tracking-application-lifecycle-events-for-android-platform">Tracking application lifecycle events on the Android platform</a> section.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 8]
```diff
@@ -81,6 +76,33 @@ RudderClient rudderClient = RudderClient.GetInstance(
...
+| `recordScreenViews`     | Boolean | Determines if the SDK will automatically capture the screen view events.                   | `false`            |
+| `autoCollectAdvertId`  | Boolean | Determines if the SDK will collect the advertisement ID.  | `false` |
+| `controlPlaneUrl`       | String  | Change this parameter **only if** you are self-hosting the control plane. Check the [Self-hosted control plane](#self-hosted-control-plane) section below for more information. The SDK will add `/sourceConfig` along with this URL to fetch the source configuration. | [https://api.rudderlabs.com](https://api.rudderlabs.com) |
+
+
+### Self-hosted control plane
+
+If you are using a device mode destination like Adjust, Firebase, etc., the Unity SDK needs to fetch the required configuration from the control plane. If you are using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to host your own Control Plane, then specify `controlPlaneUrl` in your`RudderConfigBuilder` that points to your hosted source configuration file.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:99]:
```suggestion
If you are using a device mode destination like Adjust, Firebase, etc., the Unity SDK needs to fetch the required configuration from the control plane. If you are using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to host your own Control Plane, then specify `controlPlaneUrl` in your `RudderConfigBuilder` that points to your hosted source configuration file.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx [hunk 9]
```diff
@@ -81,6 +76,33 @@ RudderClient rudderClient = RudderClient.GetInstance(
...
+
+<div class="warningBlock">
+  
+You shouldn't pass the <code class="inline-code">controlPlaneUrl</code> parameter during SDK initialization if you are using the <a href="https://app.rudderstack.com">RudderStack Cloud dashboard</a>. This parameter is supported only if you are using the open source <a href="https://rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to self-host your control plane.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx:103]:
```suggestion
You should not pass the <code class="inline-code">controlPlaneUrl</code> parameter during the SDK initialization if you are using the <a href="https://app.rudderstack.com">RudderStack Cloud dashboard</a>. This parameter is supported only if you are using the open source <a href="https://rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to self-host your control plane.
```

**Review by arnab-p — APPROVED** [review]

**Pull request merged**